### PR TITLE
Restructure k8s manifests to suport bundle generation

### PR DESCRIPTION
**Pull request by lburgazzoli**
- feat: re-structure kubernetes manifest to support bundle generation
    - the kubernetes manifest folder has been restructured and simplified
    - operators and sync configuration has been simplified to reduce the need to apply kustomize patch
    - the deployment to dev cluster has been simplified ans may be replaced by ArgoCD at some point
- feat: remove tenancy feature flag
    - as part of the simplification of the set-up the feature flag to disable multi tenancy has been removed given that it is not more needed to support non multi tenant managers
- chore: update connector example

### CODE REVIEW

**Review by valdar — APPROVED** [review]
lgtm

File: etc/examples/argocd-app.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+kind: Application
+metadata:
+  name: rhoc
+spec:
+  destination:
+    namespace: cos
+    server: https://kubernetes.default.svc
+  project: default
+  source:
+    path: etc/kubernetes/manifests/overlays/lburgazzoli
```
Comment (valdar) [inline etc/examples/argocd-app.yaml:11]: is this a leftover?

File: etc/examples/argocd-app.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    repoURL: https://github.com/lburgazzoli/bf2-cos-fleetshard
```
Comment (valdar) [inline etc/examples/argocd-app.yaml:12]: same here?

**Review by valdar — CHANGES_REQUESTED** [review]

File: etc/examples/argocd-app.yaml (hunk 1 above)
Comment (lburgazzoli) [reply etc/examples/argocd-app.yaml:11]: yes, it will be removed before marking the PR as ready

File: etc/examples/argocd-app.yaml (hunk 2 above)
Comment (lburgazzoli) [reply etc/examples/argocd-app.yaml:12]: yes, it will be removed before marking the PR as ready

File: cos-fleetshard-poms/pom.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    <parent>
+        <groupId>org.bf2</groupId>
+        <artifactId>cos-fleetshard</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>cos-fleetshard-poms</artifactId>
```
Comment (rk3rn3r) [inline cos-fleetshard-poms/pom.xml:14]: What's exactly the role of this module?

Comment (lburgazzoli) [reply cos-fleetshard-poms/pom.xml:14]: It is ti simplify the poms for the operator/sync as they share almost the same configuration for plugins and set-up for container image build.

**Review by valdar — APPROVED** [review]
lgtm

**Pull request merged**